wrappers/python/nntile/layer/gpt2_attention.py

```diff
@@ -0,0 +1,982 @@
+        )
+
+        weight_torch_np = np.empty((n_emb, 3*n_emb))
+        weight_torch_np[:, : n_emb] = to_numpy(
+            self.w_q.value
+            ).reshape(n_emb, n_emb).T
+        weight_torch_np[:, n_emb : 2*n_emb] = to_numpy(
+            self.w_k.value
+            ).reshape(n_emb, n_emb).T
+        weight_torch_np[:, 2*n_emb : 3*n_emb] = to_numpy(
+            self.w_v.value
+            ).reshape(n_emb, n_emb).T    
+        torch_layer.c_attn.weight.data = torch.tensor(
+            weight_torch_np,
+            requires_grad=True,
+        )
+
+        bias_torch_np = np.empty((3*n_emb,))
+        bias_torch_np[: n_emb] = to_numpy(
+            self.in_proj_bias_q.value
+            ).T.reshape(n_emb,)
+        bias_torch_np[n_emb : 2*n_emb] = to_numpy(
+            self.in_proj_bias_k.value
+            ).T.reshape(n_emb,)
+        bias_torch_np[2*n_emb : 3*n_emb] = to_numpy(
+            self.in_proj_bias_v.value
+            ).T.reshape(n_emb,)
+        torch_layer.c_attn.bias.data = torch.tensor(
+            bias_torch_np,
+            requires_grad=True,
+        )
+
+        torch_layer.c_proj.weight.data = torch.tensor(
+            to_numpy(self.w.value)
+            .reshape(n_emb, n_emb).T,
+            requires_grad=True,
+        )
+        torch_layer.c_proj.bias.data = torch.tensor(
+            to_numpy(self.out_proj_bias.value),
+            requires_grad=True,
+        )
+        return torch_layer
+
+    def to_torch_with_grads(self) -> GPT2Attention_torch:
+        n_embd = self.head_size * self.n_head
+        torch_layer = self.to_torch()
+        weight_torch_np = np.empty((n_embd, 3*n_embd))
+        weight_torch_np[:, : n_embd] = to_numpy(
+            self.w_q.grad
+            ).reshape(n_embd, n_embd).T
+        weight_torch_np[:, n_embd : 2*n_embd] = to_numpy(
+            self.w_k.grad
+            ).reshape(n_embd, n_embd).T
+        weight_torch_np[:, 2*n_embd : 3*n_embd] = to_numpy(
+            self.w_v.grad
+            ).reshape(n_embd, n_embd).T    
+        torch_layer.c_attn.weight.grad = torch.tensor(
+            weight_torch_np
+        )
+
+        bias_torch_np = np.empty((3*n_embd,))
+        bias_torch_np[: n_embd] = to_numpy(
+            self.in_proj_bias_q.grad
+            ).T.reshape(n_embd,)
+        bias_torch_np[n_embd : 2*n_embd] = to_numpy(
+            self.in_proj_bias_k.grad
+            ).T.reshape(n_embd,)
+        bias_torch_np[2*n_embd : 3*n_embd] = to_numpy(
+            self.in_proj_bias_v.grad
+            ).T.reshape(n_embd,)
+        torch_layer.c_attn.bias.grad = torch.tensor(
+            bias_torch_np
+        )
+
+        torch_layer.c_proj.weight.grad = torch.tensor(
+            to_numpy(self.w.grad)
+            .reshape(n_embd, n_embd).T
+        )
+        torch_layer.c_proj.bias.grad = torch.tensor(
+            to_numpy(self.out_proj_bias.grad)
+        )
+        return torch_layer
```

Comment:
no end-of-line symbol in the end of the file. Change settings of your VSCode to add EOL at the end of each file